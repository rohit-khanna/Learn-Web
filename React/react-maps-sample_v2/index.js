import React from "react";
import { render } from "react-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import "./index.css";
import App from "./src/components/App";
import ConfigureStore from "./src/redux/configureStore";
import InitialState from "./src/redux/reducers/InitialState";
import { Provider as ReduxProvider } from "react-redux";

const store = new ConfigureStore(InitialState);

render(
  <ReduxProvider store={store}>
    <App />
  </ReduxProvider>,
  document.getElementById("app")
);
