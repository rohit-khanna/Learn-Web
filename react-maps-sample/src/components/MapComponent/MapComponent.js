import React from 'react';
import { compose, lifecycle } from 'recompose';
import { withScriptjs, withGoogleMap, GoogleMap, Marker, DirectionsRenderer } from 'react-google-maps';

const MapComponent = compose(
	withScriptjs,
	withGoogleMap,
	lifecycle({
		componentDidMount() {
			//const DirectionsService = new window.google.maps.DirectionsService();
			//const { markers } = this.props;
			// DirectionsService.route(
			// 	{
			// 		origin: { lat: Number(markers[0][0]), lng: Number(markers[0][1]) },
			// 		destination: { lat: Number(markers[1][0]), lng: Number(markers[1][1]) },
			// 		//waypoints: [ { location: { lat: 41.85258, lng: -87.6514 }, stopover: false } ],
			// 		travelMode: window.google.maps.TravelMode.DRIVING
			// 	},
			// 	(result, status) => {
			// 		if (status === window.google.maps.DirectionsStatus.OK) {
			// 			this.setState({
			// 				directions: result
			// 			});
			// 		} else {
			// 			console.error(`error fetching directions ${result}`);
			// 		}
			// 	}
			// );
		}
	})
)((props) => (
	<GoogleMap
		zoom={12}
		center={
			props.markers && props.markers.length > 2 ? (
				{ lat: Number(props.markers[1][0]), lng: Number(props.markers[1][1]) }
			) : (
				{ lat: 28.7417623, lng: 77.1370109 }
			)
		}
	>
		{props.markers &&
			props.markers.map((marker, idx) => (
				<Marker key={idx} position={{ lat: Number(marker[0]), lng: Number(marker[1]) }} />
			))}
	</GoogleMap>
));

export default MapComponent;
