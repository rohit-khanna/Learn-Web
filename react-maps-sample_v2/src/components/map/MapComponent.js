import React from "react";
import BaseComponent from "../BaseComponent";
import { MapService } from "../../services";
import {
  withScriptjs,
  withGoogleMap,
  GoogleMap,
  //Marker,
  DirectionsRenderer
} from "react-google-maps";

/**
 * Inner class For Map Component
 */
class MapComponentInner extends React.Component {
  //constructor
  constructor(props) {
    super(props);
    this.state = {
      directions: ""
    };
  }

  /**
   * Lifecycle method invoked once component is mounted
   */
  async componentDidMount() {
    const { markers } = this.props;
    if (markers && markers.length > 2) {
      const [start, ...rest] = markers;
      const [end] = rest.slice(-1);

      try {
        // const directions = await MapService.FetchDirections(
        //   {
        //     lat: start[0],
        //     lng: start[1]
        //   },
        //   {
        //     lat: end[0],
        //     lng: end[1]
        //   },
        //   (markers.length > 2 ? rest.slice(0, markers.length - 2) : []).map(
        //     wayPoint => {
        //       return {
        //         lat: wayPoint[0],
        //         lng: wayPoint[1]
        //       };
        //     }
        //   )
        // );
        // this.setState({ directions });
      } catch (error) {
        alert(error);
      }
    }
  }

  /**
   * Renders the Google Map
   */
  render() {
    const { markers } = this.props;
    const { directions } = this.state;
    return (
      <GoogleMap
        zoom={12}
        center={
          markers && markers.length > 2
            ? { lat: Number(markers[1][0]), lng: Number(markers[1][1]) }
            : { lat: 28.7417623, lng: 77.1370109 }
        }
      >
        {directions && <DirectionsRenderer directions={directions} />}
      </GoogleMap>
    );
  }
}

// Wraping the inner component with Google Scripts, to setup the necesary wrappers for Maps
const MapComponent = withScriptjs(
  withGoogleMap(BaseComponent(MapComponentInner))
);

export default MapComponent;
