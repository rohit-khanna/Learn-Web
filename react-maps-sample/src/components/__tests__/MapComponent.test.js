import React from "react";
import MapComponent from "../MapComponent";
import renderer from "react-test-renderer";

test("Map Component Tests", () => {
  const component = renderer.create(
    <MapComponent
      googleMapURL="http:\\google.com"
      isMarkerShown
      loadingElement={<div style={{ height: `100%` }} />}
      containerElement={<div style={{ height: `100vh` }} />}
      mapElement={<div style={{ height: `100%` }} />}
      markers={[]}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
