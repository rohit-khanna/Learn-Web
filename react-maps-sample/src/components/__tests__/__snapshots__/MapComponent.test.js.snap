// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Component Tests 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
/>
`;
